src/screens/General/Dashboard.js

```diff
@@ -59,212 +56,219 @@ class Dashboard extends Component {
+					changeVisibility = { (val) => this.setState({ flagsVisible: val }) }
+					flagPicked = { (text) => this.flagPicked(text) }
+				/>
+				<CustomFlag
+					customFlagVisible = { customFlagVisible }
+					customFlagText = { customFlagText }
+					changeText = { (text) => this.setState({ customFlagText: text }) }
+					changeVisibility = { (val) => this.setState({ customFlagVisible: val }) }
+					flagPicked = { (text) => this.flagPicked(text) }
+				/>
 			</SafeAreaView>
 		);
 	}
 
-	renderColorPicker() {
+	renderHeader() {
+		const {
+			headerContainer,
+			headerOptionsContainer
+		} = styles;
+
+		let dashColor = { backgroundColor: this.props.dashColor };
+		let chevronColor = { color: "white" };
+
 		return (
-			<Modal visible = { this.state.colorPicker } transparent = { true }>
-				<View style = { [styles.modalBackground, { backgroundColor: "transparent" }] }>
-					<ColorPicker
-						defaultColor = "#21252b"
-						oldColor = { this.props.dashColor }
-						onColorSelected = { color => this.colorPicked(color) }
-						style = { [styles.modalContent, { backgroundColor: "black" }] }
+			<View style = { [headerContainer, dashColor ] }>
+				{ this.renderGreeting() }
+				<View style = { headerOptionsContainer }>
+					{ this.renderHeaderFlag() }
+					<FontAwesomeIcon
+						style = { chevronColor }
+						name = "chevron-down"
+						onPress = { () => this.setState({ colorPickerVisible: true }) }
+						size = { 15 }
 					/>
 				</View>
-			</Modal>
+			</View>
 		);
 	}
 
-	renderHeader() {
+	renderGreeting() {
 		const {
-			greetingContainerStyle
+			textColor,
+			greetingContainer
 		} = styles;
 
+		const date = new Date();
+		let time = date.getHours();
+		let day = date.getDate();
+		let month = date.getMonth();
+		let greeting = time >= 12 ? "Good evening" : "Good morning";
+
 		return (
-			<View style = { [greetingContainerStyle, { backgroundColor: this.props.dashColor, justifyContent: "center" }] }>
-				{ this.greeting() }
-				<View style = {{ justifyContent: "space-evenly", alignItems: "center", paddingLeft: "2%", paddingRight: "2%" }}>
-					{ this.flag() }
-					<View style = {{ justifyContent: "center", alignItems: "center" }}>
-						<FontAwesomeIcon style = {{ color: "white" }} name = "chevron-down" onPress = { () => this.setState({ colorPicker: true }) } size = { 15 } />
-					</View>
-				</View>
+			<View style = { greetingContainer }>
+				<Text style = { [textColor, { fontSize: 20 }] }>{ greeting }, { this.props.firstName }.</Text>
+				<Text style = { textColor }>Today is { months[month] } { day }</Text>
 			</View>
 		);
 	}
 
-	renderEvents() {
+	renderHeaderFlag() {
 		return (
-			<View>
-				{ this.getFormattedEventList() }
-			</View>
+			<TouchableOpacity onPress = { () => this.setState({ flagsVisible: !this.state.flagsVisible }) } >
+				<Flag
+					type = "flat"
+					code = { this.props.flag }
+					size = { 32 }
+				/>
+			</TouchableOpacity>
 		);
 	}
 
+	 flagPicked(flag) {
```

Comment:
All the flag stuff shouldn't be in this component. If you don't want to deal with it now, open an issue to make that it's own component